sympy/tensor/tensor.py

```diff
@@ -4861,3 +4861,23 @@ def _expand(expr, **kwargs):
         return expr._expand(**kwargs)
     else:
         return expr.expand(**kwargs)
+
+
+def get_postprocessor(cls):
+    def _postprocessor(expr):
+        tens_class = {Mul: TensMul, Add: TensAdd}[cls]
+        if any(isinstance(a, TensExpr) for a in expr.args):
+            return tens_class(*expr.args)
+        else:
+            return expr
+
+    return _postprocessor
+
+"""
+The following makes sure that if a user tries to put a tensor in a Mul, it
+automatically gets converted to a TensMul (see github issue #25051). The
+conversion already seems to work for TensAdd, so we don't do anything for that.
+"""
```

Comment:
Use # for comments. You also don't need to mention all these details in a comment. Just the first sentence here is enough, or even nothing at all, since people can just look up what the constructor postprocessor is in basic.py if they don't know what this code does.

Reply:
Okay, I've removed that comment.